### PR TITLE
Update test names from "Electron" to "Core "

### PR DESCRIPTION
**Pull request by SakshiS-harma**
Update test naming convention from Electron to Core to avoid confusions

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -71,13 +71,13 @@ jobs:
       - name: Compile and Download
         run: yarn npm-run-all --max_old_space_size=4095 -lp compile "electron x64" # {{SQL CARBON EDIT}} Remove unused options playwright-install download-builtin-extensions
 
-      - name: Run Unit Tests (Electron)
+      - name: Run Core Unit Tests
```
Comment (Charles-Gagnon) [inline .github/workflows/ci.yml:74]: Add `{{SQL CARBON EDIT}} Rename to core for better clarity` to lines changed in this file

File: build/azure-pipelines/darwin/product-build-darwin-test.yml [hunk 2]
```diff
@@ -157,7 +157,7 @@ steps:
   - script: |
       set -e
       ./scripts/test.sh --build --tfs "Unit Tests"
-    displayName: Run unit tests (Electron)
+    displayName: Run core unit tests
```
Comment (Charles-Gagnon) [inline build/azure-pipelines/darwin/product-build-darwin-test.yml:160]: Remove changes to any of the product-build pipelines that don't start with sql, we don't use those. 

(and you're missing sql-product-build-darwin)

File: .github/workflows/ci.yml (hunk 1 above)
Comment (SakshiS-harma) [reply .github/workflows/ci.yml]: Updated.

File: build/azure-pipelines/darwin/product-build-darwin-test.yml (hunk 2 above)
Comment (SakshiS-harma) [reply build/azure-pipelines/darwin/product-build-darwin-test.yml]: Done.

File: build/azure-pipelines/darwin/sql-product-build-darwin.yml [hunk 3]
```diff
@@ -122,7 +122,7 @@ steps:
       INTEGRATION_TEST_ELECTRON_PATH="$APP_ROOT/$APP_NAME/Contents/MacOS/Electron" \
       VSCODE_REMOTE_SERVER_PATH="$(agent.builddirectory)/azuredatastudio-reh-darwin" \
       ./scripts/test-integration.sh --build --tfs "Integration Tests"
-    displayName: Run integration tests (Electron)
+    displayName: Run core integration tests # {{SQL CARBON EDIT}} Rename to core for clarity
```
Comment (Charles-Gagnon) [inline build/azure-pipelines/darwin/sql-product-build-darwin.yml:125]: Remove sql carbon edit for any `sql-*` yml files - those are ones that we own completely so no need to mark them as changed.

Comment (SakshiS-harma) [reply build/azure-pipelines/darwin/sql-product-build-darwin.yml]: Done.

**Review by Charles-Gagnon — APPROVED** [review]

**Pull request merged**